### PR TITLE
feat: Add new terraform project for e2e test resources

### PR DESCRIPTION
**Pull request by hessjcg**
This is a copy of the resource provisioning code from `testinfra/*.tf` with this modification: IAM and permissions 
related terraform code is in the `infra/permissions` terraform project. 

Related to #65

### CODE REVIEW

File: infra/resources/artifacts.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2022 Google LLC
```
Comment (enocom) [inline infra/resources/artifacts.tf:2]: 2023. Fix this in the past PR if we missed it too please.

Comment (enocom) [reply infra/resources/artifacts.tf:2]: Also update https://github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/blob/main/.github/header-checker-lint.yml to include .tf (and whatever other file types we might need in this repo).

Comment (enocom) [reply infra/resources/artifacts.tf:2]: Ditto below.

File: infra/resources/gke_cluster.tf [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
+    service_account = var.nodepool_serviceaccount_email
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform",
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/sqlservice.admin",
```
Comment (enocom) [inline infra/resources/gke_cluster.tf:78]: You might need sqlservice.login if you're going to do auto IAM authn.

**Review by enocom — CHANGES_REQUESTED** [review]

File: infra/resources/artifacts.tf (hunk 1 above)
Comment (hessjcg) [reply infra/resources/artifacts.tf:2]: Fixed. I updated all the files here to 2023. The infra/permissions are also copyright 2023.

File: infra/resources/gke_cluster.tf (hunk 2 above)
Comment (hessjcg) [reply infra/resources/gke_cluster.tf:78]: Good note. I'll add that later when we write the the IAMAuthN test.

**Review by enocom — APPROVED** [review]

**Pull request merged**